src/util/triplea/MapXMLCreator/MapXMLCreator.java

```diff
@@ -0,0 +1,839 @@
+		size.height -= 100;
+		spTaNotes.setMinimumSize(size);
+		
+		GridBagConstraints gbc_lFirstRow = new GridBagConstraints();
+		gbc_lFirstRow.insets = new Insets(0, 0, 5, 0);
+		gbc_lFirstRow.gridy = 0;
+		gbc_lFirstRow.gridx = 0;
+		gbc_lFirstRow.anchor = GridBagConstraints.NORTH;
+		gbc_lFirstRow.weightx = 1.0;
+		gbc_lFirstRow.weighty = 1.0;
+		gbc_lFirstRow.gridwidth = 3;
+		m_stepActionPanel.add(spTaNotes, gbc_lFirstRow);
+
+		GridBagConstraints gbc_lSecondRow = (GridBagConstraints) gbc_lFirstRow.clone();
+		gbc_lSecondRow.weighty = 0.0;
+		gbc_lSecondRow.insets = new Insets(0, 0, 0, 0);
+		gbc_lSecondRow.gridwidth = 1;
+		gbc_lSecondRow.gridx = 1;
+		gbc_lSecondRow.gridy = 1;
+		
+		final JButton bPreviewHTML = new JButton("Preview HTML");
+		bPreviewHTML.setPreferredSize(new Dimension(300, 30));
+		bPreviewHTML.setAction(new AbstractAction("Preview HTML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				showHTML(MapXMLHelper.s_notes, "HTML Preview");
+			}
+		});
+		m_stepActionPanel.add(bPreviewHTML, gbc_lSecondRow);
+
+		GridBagConstraints gbc_lThirdRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lThirdRow.gridy = 2;
+		m_stepActionPanel.add(new JLabel("<html><big>Congratulation!</big></html>"), gbc_lThirdRow);
+		
+		GridBagConstraints gbc_lFourthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFourthRow.gridy = 3;
+		m_stepActionPanel.add(new JLabel("<html><nobr>You have completed all the steps for creating the map XML.</nobr></html>"), gbc_lFourthRow);
+		
+		final JButton bSave = new JButton("Save Entries to XML");
+		bSave.setPreferredSize(new Dimension(600, 35));
+		bSave.setAction(new AbstractAction("Save Map XML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				MapXMLHelper.saveXML();
+			}
+		});
+		GridBagConstraints gbc_lFifthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFifthRow.insets = new Insets(5, 0, 0, 0);
+		gbc_lFifthRow.gridy = 4;
+		m_stepActionPanel.add(bSave, gbc_lFifthRow);
+	}
+	
+	protected void layoutTabbedPaneWith(final Set<String> keySet)
+	{
+		m_stepActionPanel.setLayout(new BorderLayout());
+		final JTabbedPane tabbedPane = new JTabbedPane(JTabbedPane.SCROLL_TAB_LAYOUT);
+		for (final String key : keySet)
+		{
+			final JPanel innerTabPane = new JPanel();
+			tabbedPane.addTab(key, innerTabPane);
+			ProductionFrontiersPanel.layout(this, innerTabPane, key);
+		}
+		m_stepActionPanel.add(tabbedPane, BorderLayout.CENTER);
+	}
+	
+	
+	/**
+	 * @
+	 * 
+	 * @see HtmlEditorKitTest {@link}https://github.com/KaBoKu/PI/blob/master/WPProjekt/src/kus/swing/HtmlEditorKitTest.java
+	 */
+	public void showHTML(final String htmlString, final String title)
+	{
+		SwingUtilities.invokeLater(new Runnable()
+		{
+			public void run()
+			{
+				// create jeditorpane

```

Comment:
Yeah, to read this code block I'm having to really block out what the comments say. Reading comment + redundant line of code is a bit much, particularly since a reader is aware that any comment can and is likely to be out of date.